server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -378,7 +399,9 @@ def generate_token(
     ) -> Tuple[List[Generation], Optional[FlashCausalLMBatch]]:
         # Shortcut when batch_size == 1
         if len(batch) == 1:
-            # No need to slice this down
+            input_ids = batch.input_ids[0].view(-1)
+            # Slice to remove extra padding
+            # past_key_values = batch.past_key_values[:, :batch.input_lengths[0]] if batch.past_key_values is not None else None
```

Comment:
The modeling code just uses offsets so no need to slice...

Reply:
Yes I forgot to cleanup again...